main.cpp

```diff
@@ -120,6 +121,17 @@ bool Dishwasher::lockDoor()
     return true;
 }
 
+void Dishwasher::doesSomethingInteresting()
+{
+    int inc = 0;
+    std::cout << "'inc' initialized with value " << inc << std::endl;
+    for (int i=0; i < 5; i++) {        
+        inc+= 5;
+        std::cout << "'inc' incremented by 5 --> " << inc << std::endl;
+    }
```

Comment:
1) Curly braces go on their own line.
2) Everywhere that you have `i++`, `i--`, `i -= 1;` or `i += 1;`, you should use `pre-increment/decrement`. 

You are currently doing `post-increment/decrement`.  
you can think of post- and pre- increment like this:
```
int preIncrement(int& i)
{
    i += 1;
    return i; //both the returned value AND i have changed
}
​
int postIncrement(int& i)
{
    int oldVal = i;
    i += 1;
    return oldVal; //i has changed but the value being returned does not reflect this change.
}
```
more information can be found here explaining why `pre-increment/decrement` is preferred over `post-increment/decrement`: https://en.cppreference.com/w/cpp/language/operator_incdec

3) add space around `+=` so your code is easier to read.
- [ ] revise

Reply:
I confirm the change from post increment/decrement to pre-increment/decrement however I am not entirely sure why it is preferable - is it a performance improvement because there are less instructions from the compiler (no copy ?)

thanks